### PR TITLE
feat(drawer): create drawer component

### PR DESCRIPTION
**Pull request by damianOfKyan**
### Proposed behaviour
Drawer component, a simple two container layout that allows for left hand side container to expand and collapse as per user choice/settings white the right hand side container adjusts reflows its content based on available space.

drawer

### Checklist

- [x] Release notes (Conventional Commits) 
<del>- [ ] Unit tests
- [x] Cypress automation tests
- [x] Storybook added or updated
<del>- [ ] Theme support
- [x] Typescript `d.ts` file added or updated

### CODE REVIEW

File: src/components/drawer/drawer.component.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  const closedClass = firstOpen ? '' : 'closed';
+
+  return (
+    <StyledDrawerWrapper
+      isExpanded={ isExpanded }
+      { ...props }
+    >
+      <StyledDrawerContent
+        expandedWidth={ expandedSettings ? expandedSettings.maxWidth : '40%' }
```
Comment (jamime) [inline src/components/drawer/drawer.component.js:40]: I think we should change this interface. The DLS specified a few explicit width options. 
See other suggestions below for complete example.
```suggestion
        expandedWidth={ expandedWidth }
```

File: src/components/drawer/drawer.component.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  /** Sets the expansion state of the Drawer if component is meant to be used as controlled */
+  expanded: PropTypes.bool,
+  /** Callback fired when expansion state changes, onChange(event: object, isExpanded: boolean) */
+  onChange: PropTypes.func,
+  sidebar: PropTypes.node,
+  expandedSettings: PropTypes.object
+};
+
+Drawer.defaultProps = {
+  defaultIsExpanded: false
```
Comment (jamime) [inline src/components/drawer/drawer.component.js:68]:
```suggestion
  defaultIsExpanded: false,
  expandedWidth: '40%'
```

File: src/components/drawer/drawer.component.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+Drawer.displayName = 'Drawer';
```
Comment (jamime) [inline src/components/drawer/drawer.component.js:71]: This should be the default, line is probably not needed?

File: src/components/drawer/drawer.component.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+Drawer.propTypes = {
+  children: PropTypes.node,
+  /** Set the default state of expansion of the Drawer if component is meant to be used as uncontrolled */
+  defaultIsExpanded: PropTypes.bool,
+  /** Sets the expansion state of the Drawer if component is meant to be used as controlled */
+  expanded: PropTypes.bool,
+  /** Callback fired when expansion state changes, onChange(event: object, isExpanded: boolean) */
+  onChange: PropTypes.func,
+  sidebar: PropTypes.node,
+  expandedSettings: PropTypes.object
```
Comment (jamime) [inline src/components/drawer/drawer.component.js:64]: Replace the values with the real ones from DLS.
```suggestion
  expandedWidth: PropTypes.oneOf(['40%', '50%', '60%'])
```

File: src/components/drawer/drawer.component.js (hunk 1 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js]: fixed

File: src/components/drawer/drawer.component.js (hunk 3 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js]: yes

File: src/components/drawer/drawer.component.js (hunk 4 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js]: after chatting to Harpal, we decided to go % and px way., i set the propType to string.

File: src/components/drawer/drawer.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+<Meta title="Design System/Drawer" />
+
+# Drawer
+
+### Basic usage:
+
+<Preview>
+  <Story name="default" parameters={{ info: { disable: true }}} >
+  {() => {
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.stories.mdx]: indeed. thanks!

File: src/components/drawer/drawer.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import { Meta, Props, Preview, Story } from '@storybook/addon-docs/blocks';
+import Drawer from '.';
+import { useState } from 'react';
+
+<Meta title="Design System/Drawer" />
+
+# Drawer
+
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.stories.mdx:11]: added

File: src/components/drawer/drawer.d.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+
+export interface DrawerPropTypes {
+  children: React.FunctionComponent | React.ComponentClass;
+  defaultIsExpanded?: boolean;
+  expanded?: boolean;
+  onChange?: () => void;
+  sidebar?: () => void;
+  expandedSettings?: () => void;
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.d.ts:9]: now using as a string

File: src/components/drawer/drawer.stories.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+import { withKnobs } from '@storybook/addon-knobs';
+import Drawer from '.';
+
+export default {
+  title: 'Test/Drawer',
+  component: Drawer,
+  decorators: [withKnobs]
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.stories.js]: removed

File: src/components/drawer/drawer.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+              <li>link b</li>
+              <li>link c</li>
+            </ul>
+          ) }
+        >
+          content body content body content body content body content body content body content body
+        </Drawer>
+      </div>
+    )
+  }}
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.stories.mdx]: fixed

File: src/components/drawer/drawer.component.js (hunk 2 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js]: fixed

File: src/components/drawer/drawer.d.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+
+export interface DrawerPropTypes {
+  children: React.FunctionComponent | React.ComponentClass;
+  defaultIsExpanded?: boolean;
+  expanded?: boolean;
+  onChange?: () => void;
+  sidebar?: () => void;
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.d.ts]: fixed

File: src/components/drawer/drawer.component.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        {children}
+      </StyledDrawerChildren>
+    </StyledDrawerWrapper>
+  );
+};
+
+Drawer.propTypes = {
+  children: PropTypes.node,
+  /** Set the default state of expansion of the Drawer if component is meant to be used as uncontrolled */
+  defaultIsExpanded: PropTypes.bool,
```
Comment (jamime) [inline src/components/drawer/drawer.component.js:64]: This should match the controlled prop `expanded`.
```suggestion
  defaultExpanded: PropTypes.bool,
```

File: src/components/drawer/drawer.component.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+  /** Callback fired when expansion state changes, onChange(event: object, isExpanded: boolean) */
+  onChange: PropTypes.func,
+  /* Sidebar object either html or react component */
+  sidebar: PropTypes.node,
+  /* The (% or px) width of the expanded sizebar  */
+  expandedWidth: PropTypes.string
+};
+
+Drawer.defaultProps = {
+  defaultIsExpanded: false,
```
Comment (jamime) [inline src/components/drawer/drawer.component.js:76]: I don't think we need this default prop
```suggestion
```

File: src/components/drawer/drawer.style.js [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+    }
+  }
+`;
+
+const StyledDrawerWrapper = styled.div`
+  display: flex;
+  height: 100%;
+
+  ${StyledDrawerContent} {
+    ${StyledIconButton} {
```
Comment (jamime) [inline src/components/drawer/drawer.style.js:99]: Can we move this to `StyledIconButton` on line 4?

File: src/components/drawer/drawer.component.js [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+  expanded,
+  onChange,
+  children,
+  expandedWidth,
+  sidebar,
+  ...props
+}) => {
+  const isControlled = expanded !== undefined;
+  const [isExpandedInternal, setIsExpandedInternal] = useState(defaultIsExpanded || false);
+  const [firstOpen, setFirstOpen] = useState(true);
```
Comment (jamime) [inline src/components/drawer/drawer.component.js:24]:
```suggestion
  const [isFirstOpen, setFirstOpen] = useState(true);
```

File: src/components/drawer/drawer.component.js (hunk 11 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js]: fixed.

File: src/components/drawer/drawer.component.js (hunk 12 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js]: correct, removed.

File: src/components/drawer/drawer.style.js (hunk 13 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.style.js]: updated.

File: src/components/drawer/drawer.component.js (hunk 14 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js]: fixed

File: cypress/features/regression/test/drawer.feature [hunk 15]
```diff
@@ -0,0 +1,20 @@
+Feature: Drawer component
+  I want to test Drawer component
+
+  Background: Open Drawer component page
+    Given I open basic Test "Drawer" component page
+
+  @positive
+  Scenario: Expanding Drawer
+    Given I click on Drawer arrow
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/drawer.feature:9]:
```suggestion
    When I click on Drawer arrow
```

File: cypress/support/step-definitions/drawer-steps.js [hunk 16]
```diff
@@ -0,0 +1,29 @@
+import {
+  drawerToggle,
+  drawerSidebar,
+  drawerSidebarContent,
+} from '../../locators/drawer';
+
+When('I click on Drawer arrow', () => {
+  drawerToggle().first().click();
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/drawer-steps.js:8]: try to use `for` loop to pass how many times you want to click this arrow. In feature file you have doubled steps in When/ And

File: cypress/features/regression/test/drawer.feature [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+    Then sidebar should have class open
+      And toggle icon switched orientation to open
+      And sidebar text is visible
+
+  @positive
+  Scenario: Collapsing Drawer
+    Given I click on Drawer arrow
+      And I click on Drawer arrow
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/drawer.feature:17]: try to create step like `I click on Drawer arrow {int} times` . Then you can remove `And` step and in the `Given` pass 2

File: cypress/locators/drawer/locators.js [hunk 18]
```diff
@@ -0,0 +1,5 @@
+// component preview locators
+export const DRAWER = '[data-element="drawer"]';
+export const DRAWER_TOGGLE = '[data-element="drawer-toggle"]';
+export const DRAWER_SIDEBAR = '[data-element="drawer"] div:nth-child(1)';
```
Comment (KatarzynaQA) [inline cypress/locators/drawer/locators.js:4]: move it to `index.js` file as
`export const drawerSidebar = () => cy.iFrame(DRAWER).find(div:nth-child(1));`

File: cypress/locators/drawer/locators.js [hunk 19]
```diff
@@ -0,0 +1,5 @@
...
+export const DRAWER_SIDEBAR_CONTENT = `${DRAWER_SIDEBAR} > div`;
```
Comment (KatarzynaQA) [inline cypress/locators/drawer/locators.js:5]: move it to index.js file as
`export const drawerSidebarContent = () => drawerSidebar().find('div')`

File: cypress/support/step-definitions/drawer-steps.js [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+});
+
+Then('sidebar should have class {word}', (className) => {
+  drawerSidebar().should('have.class', className);
+});
+
+Then('sidebar text is visible', () => {
+  drawerSidebarContent().find('li:nth-child(1)').should('have.text', 'link a').and('be.visible');
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/drawer-steps.js:16]: add Enter before `.and` line
```suggestion
  drawerSidebarContent().find('li:nth-child(1)').should('have.text', 'link a')
    .and('be.visible');
```

File: cypress/support/step-definitions/drawer-steps.js [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+});
+
+Then('sidebar text is not visible', () => {
+  drawerSidebarContent().find('li:nth-child(1)').should('have.text', 'link a').and('not.be.visible');
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/drawer-steps.js:20]:
```suggestion
  drawerSidebarContent().find('li:nth-child(1)').should('have.text', 'link a')
    .and('not.be.visible');
```

File: cypress/features/regression/test/drawer.feature (hunk 15 above)
Comment (damianOfKyan) [reply cypress/features/regression/test/drawer.feature]: fixed 4d965d0

File: cypress/features/regression/test/drawer.feature (hunk 17 above)
Comment (damianOfKyan) [reply cypress/features/regression/test/drawer.feature]: fixed 4d965d0

File: cypress/locators/drawer/locators.js (hunk 19 above)
Comment (damianOfKyan) [reply cypress/locators/drawer/locators.js]: fixed 4d965d0

File: cypress/support/step-definitions/drawer-steps.js (hunk 21 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: fixed 4d965d0

File: cypress/support/step-definitions/drawer-steps.js (hunk 20 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: fixed 4d965d0

File: cypress/locators/drawer/locators.js (hunk 18 above)
Comment (damianOfKyan) [reply cypress/locators/drawer/locators.js]: fixed 4d965d0

File: cypress/support/step-definitions/drawer-steps.js (hunk 16 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: fixed 4d965d0

File: cypress/support/step-definitions/drawer-steps.js [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+    drawerToggle().first().click();
+  }
+});
+
+Then('sidebar should have class {word}', (className) => {
+  drawerSidebar().should('have.class', className);
+});
+
+Then('sidebar text is visible', () => {
+  drawerSidebarContent().find('li:nth-child(1)').should('have.text', 'link a')
```
Comment (DlgSHi) [inline cypress/support/step-definitions/drawer-steps.js:18]: 1. Could we move this to index? And after that use it passing the index of element?
```suggestion
  drawerSidebarContentInnerElement(1).should('have.text', 'link a')
```
2. What about verification of `link b` and `link c`?

File: cypress/locators/drawer/index.js [hunk 23]
```diff
@@ -0,0 +1,9 @@
+import {
```
Comment (DlgSHi) [inline cypress/locators/drawer/index.js:1]: We could move mentioned below locator into this file:
```suggestion
export const drawerSidebarContentInnerElement = (index) => drawerSidebarContent().find(`li:nth-child(${index})`)
```

**Comment by harpalsingh** [comment]
I'm going to have a further look at the animation and see how it can be possibly improved as it feels a bit clunky at the moment, but also when you close the drawer, we don't get a clean fade out of content, it just dissapears? with larger content that will look a bit strange, can we have the content fade out nicely in timing?

**Comment by harpalsingh** [comment]
We also discussed being able to resize the drawer based on browser resolution if required, was this not added?

File: cypress/support/step-definitions/drawer-steps.js (hunk 22 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: fixed 07d78ca

File: cypress/locators/drawer/index.js (hunk 23 above)
Comment (damianOfKyan) [reply cypress/locators/drawer/index.js:1]: fixed 07d78ca

**Comment by damianOfKyan** [comment]
> I'm going to have a further look at the animation and see how it can be possibly improved as it feels a bit clunky at the moment, but also when you close the drawer, we don't get a clean fade out of content, it just dissapears? with larger content that will look a bit strange, can we have the content fade out nicely in timing?

For me animation is smooth, sometimes on first pass its bit laggy.

**Comment by damianOfKyan** [comment]
> We also discussed being able to resize the drawer based on browser resolution if required, was this not added?

Drawer will fit whatever container it is put into and fill 100% height and width. It works on inheritance, so if parent component does not have 100% width it will auto height.

File: src/components/drawer/drawer.style.js [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    100% {
+      display: none;
+      opacity: 0;
+    }
+  }
+
+  ${StyledDrawerSidebar} {
+    display: none;
+    opacity 0;
```
Comment (mkrds) [inline src/components/drawer/drawer.style.js:46]: These styles should be declared in `StyledDrawerSidebar`, avoid cascading if it's not beneficial in any way

**Comment by mkrds** [comment]
My 2 cents regarding animating the Drawer:
Not sure if using `linear` here is a good idea - animation feels weird. 
Also to avoid having an `animationProp` and to achieve good looking animations for all dimensions we could measure the distance which Drawer will travel during the expansion and apply properly calculated animation properties

**Comment by mkrds** [comment]
Also it would be nice, not necessary though if you could put something more visually pleasing as a sidebar content, unstyled html actually looks like something is broken there

File: src/components/drawer/drawer.style.js (hunk 24 above)
Comment (mkrds) [reply src/components/drawer/drawer.style.js:46]: Also, there's a missing colon `:` in `opacity 0;` here

File: src/components/drawer/drawer.component.js [hunk 25]
```diff
@@ -0,0 +1,84 @@
...
+  /** Sets the expansion state of the Drawer if component is meant to be used as controlled */
+  expanded: PropTypes.bool,
+  /** Callback fired when expansion state changes, onChange(event: object, isExpanded: boolean) */
+  onChange: PropTypes.func,
+  /* Sidebar object either html or react component */
+  sidebar: PropTypes.node,
+  /* The (% or px) width of the expanded sizebar  */
+  expandedWidth: PropTypes.string,
+  /** Duration of a animation */
+  animationSpeed: PropTypes.string
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js:77]: fixed

File: src/components/drawer/drawer.stories.js [hunk 26]
```diff
@@ -0,0 +1,82 @@
...
+          <li>link c</li>
+        </ul>
+      ) }
+    >
+      content body content body content body content body content body content body content body
+    </Drawer>
+  );
+};
+
+Controlled.story = {
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.stories.js:74]: removed, indeed it was not needed.

File: src/components/drawer/drawer.stories.mdx [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+### Uncontrolled usage:
+
+<Preview>
+  <Story name="default" parameters={{ info: { disable: true }}} >
+    return (
+      <Drawer
+        expandedWidth='40%'
+        animationSpeed={ 500 }
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.stories.mdx]: fixed

File: src/components/drawer/drawer.stories.js [hunk 28]
```diff
@@ -0,0 +1,82 @@
...
+        </ul>
+      ) }
+    >
+      content body content body content body content body content body content body content body
+    </Drawer>
+  );
+};
+
+basic.story = {
+  name: 'Basic',
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.stories.js:39]: indeed, removed.

File: src/components/drawer/drawer.style.js [hunk 29]
```diff
@@ -0,0 +1,111 @@
...
+  }
+`;
+
+const StyledIconButton = styled(IconButton)`
+  float: right;
+  margin-top: 7px;
+  margin-right: 8px;
+
+  ${({ isExpanded }) => isExpanded && css`
+    transition: margin-right ${({ animationSpeed }) => animationSpeed} linear;
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.style.js]: fixed

File: src/components/drawer/drawer.style.js [hunk 30]
```diff
@@ -0,0 +1,111 @@
+import styled, { css } from 'styled-components';
+import IconButton from '../icon-button';
+
+const StyledDrawerChildren = styled.div`
+  flex: 1;
+  margin-left: 1px;
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.style.js:6]: it stays due to flexbox alignment the content of the right hand side container would be on top of the divider line.

File: src/components/drawer/drawer.style.js [hunk 31]
```diff
@@ -0,0 +1,111 @@
...
+      display: block;
+      opacity: 0;
+      animation: sidebar-hidden ${({ animationSpeed }) => animationSpeed} linear;
+    }
+  }
+`;
+
+const StyledIconButton = styled(IconButton)`
+  float: right;
+  margin-top: 7px;
```
[the remark this answers is not in the document]
Comment (damianOfKyan) [reply src/components/drawer/drawer.style.js:87]: I hear you, but due to size of the icon and height of the button this margin lets me achieve same distance as per designs.

File: src/components/drawer/drawer.style.js (hunk 24 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.style.js]: fixed

**Comment by damianOfKyan** [comment]
> My 2 cents regarding animating the Drawer:
> Not sure if using `linear` here is a good idea - animation feels weird.
> Also to avoid having an `animationProp` and to achieve good looking animations for all dimensions we could measure the distance which Drawer will travel during the expansion and apply properly calculated animation properties

Animations were updated to ease-in-out.

File: cypress/support/step-definitions/drawer-steps.js [hunk 32]
```diff
@@ -0,0 +1,53 @@
...
+
+Then('toggle icon switched orientation to open', () => {
+  drawerToggle().should('have.css', 'transform', 'matrix(-1, 0, 0, 1, 0, 0)');
+});
+
+Then('toggle icon switched orientation to closed', () => {
+  drawerToggle().should('have.css', 'transform', 'none');
+});
+
+// When('change expandedWidth to {word}', (width) => {
```
Comment (jamime) [inline cypress/support/step-definitions/drawer-steps.js:43]: Can we use or remove this?

File: src/components/drawer/drawer.component.js [hunk 33]
```diff
@@ -0,0 +1,85 @@
...
+      data-element='drawer'
+      { ...props }
+    >
+      <StyledDrawerContent
+        expandedWidth={ expandedWidth }
+        animationSpeed={ animationSpeed }
+        className={ isExpanded ? 'open' : closedClass }
+      >
+        <StyledIconButton
+          data-element='drawer-toggle'
```
Comment (jamime) [inline src/components/drawer/drawer.component.js:46]: I think we should look to introduce some `aria` roles here to indicate that the button controls the menu.

image

image

Details here https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html

File: src/components/drawer/drawer.d.ts [hunk 34]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+
+export interface DrawerPropTypes {
+  children: React.FunctionComponent | React.ComponentClass;
+  defaultExpanded?: boolean;
+  expanded?: boolean;
+  onChange?: () => void;
+  sidebar?: React.ReactNode;
+  expandedSettings?: string;
+  animationSpeed?: number;
```
Comment (jamime) [inline src/components/drawer/drawer.d.ts:10]:
```suggestion
  animationSpeed?: string;
```

This was changed to a string e.g. `0.4s`.

File: src/components/drawer/drawer.d.ts [hunk 35]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+
+export interface DrawerPropTypes {
+  children: React.FunctionComponent | React.ComponentClass;
+  defaultExpanded?: boolean;
+  expanded?: boolean;
+  onChange?: () => void;
```
Comment (jamime) [inline src/components/drawer/drawer.d.ts:7]: We're missing the callback signature here `e` & `isExpanded`.

File: src/components/drawer/drawer.d.ts [hunk 36]
```diff
@@ -0,0 +1,15 @@
...
+  sidebar?: React.ReactNode;
+  expandedSettings?: string;
```
Comment (jamime) [inline src/components/drawer/drawer.d.ts:9]:
```suggestion
```
Doesn't appear to be used

File: cypress/support/step-definitions/drawer-steps.js (hunk 32 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: removed, it was not needed as two steps below replaced that one.

File: src/components/drawer/drawer.component.js (hunk 33 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js:47]: good call, added aria-expanded

File: src/components/drawer/drawer.d.ts (hunk 34 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.d.ts]: updated to string

File: src/components/drawer/drawer.d.ts (hunk 35 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.d.ts]: added signature

File: src/components/drawer/drawer.d.ts (hunk 36 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.d.ts]: yeah no longer used, removed

File: src/components/drawer/drawer.component.js (hunk 33 above)
Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js:52]: added aria-label and role.

File: cypress/features/regression/test/drawer.feature [hunk 37]
```diff
@@ -0,0 +1,43 @@
...
+  @positive
+  Scenario: Collapsing Drawer
+    When I click on Drawer arrow 2 times
+    Then sidebar should have class closed
+      And toggle icon switched orientation to closed
+      And sidebar text is not visible
+
+  @positive
+  Scenario Outline: Set animationSpeed to <animationSpeed>
+    When I set animationSpeed to "<animationSpeed>"
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/drawer.feature:23]:
```suggestion
    Given I set animationSpeed to "<animationSpeed>"
```

File: cypress/features/regression/test/drawer.feature [hunk 38]
```diff
@@ -0,0 +1,43 @@
...
+    Given I click on Drawer arrow 1 time
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/drawer.feature:24]:
```suggestion
    When I click on Drawer arrow 1 time
```

File: cypress/features/regression/test/drawer.feature [hunk 39]
```diff
@@ -0,0 +1,43 @@
...
+    Examples:
+      | animationSpeed | cssAnimationSpeed |
+      | 500ms          | 0.5s              |
+      | 100ms          | 0.1s              |
+      | 0.5s           | 0.5s              |
+      | 0.1s           | 0.1s              |
+
+  @positive
+  Scenario Outline: Set expandedWidth to <expandedWidth>
+    When I set expandedWidth to "<expandedWidth>"
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/drawer.feature:35]:
```suggestion
    Given I set expandedWidth to "<expandedWidth>"
```

File: cypress/features/regression/test/drawer.feature [hunk 40]
```diff
@@ -0,0 +1,43 @@
...
+    Given I click on Drawer arrow 1 time
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/drawer.feature:36]:
```suggestion
    When I click on Drawer arrow 1 time
```

File: cypress/support/step-definitions/drawer-steps.js [hunk 41]
```diff
@@ -0,0 +1,49 @@
...
+});
+
+Then('sidebar should have class {word}', (className) => {
+  drawerSidebar().should('have.class', className);
+});
+
+Then('sidebar text is visible', () => {
+  drawerSidebarContentInnerElement(1).should('have.text', 'link a')
+    .and('be.visible');
+  drawerSidebarContentInnerElement(2).should('have.text', 'link b')
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/drawer-steps.js:20]:
```suggestion
  drawerSidebarContentInnerElement(positionOfElement('third')).should('have.text', 'link b')
    .and('be.visible');
```

File: cypress/support/step-definitions/drawer-steps.js [hunk 42]
```diff
@@ -0,0 +1,49 @@
...
+    .and('be.visible');
+  drawerSidebarContentInnerElement(3).should('have.text', 'link c')
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/drawer-steps.js:22]:
```suggestion
  drawerSidebarContentInnerElement(positionOfElement('fourth')).should('have.text', 'link c')
```

File: cypress/support/step-definitions/drawer-steps.js [hunk 43]
```diff
@@ -0,0 +1,49 @@
...
+    .and('be.visible');
+});
+
+Then('sidebar text is not visible', () => {
+  drawerSidebarContentInnerElement(1).should('have.text', 'link a')
+    .and('not.be.visible');
+  drawerSidebarContentInnerElement(2).should('have.text', 'link b')
+    .and('not.be.visible');
+  drawerSidebarContentInnerElement(3).should('have.text', 'link c')
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/drawer-steps.js:31]:
```suggestion
  drawerSidebarContentInnerElement(positionOfElement('fourth')).should('have.text', 'link c')
```

File: cypress/support/step-definitions/drawer-steps.js [hunk 44]
```diff
@@ -0,0 +1,49 @@
...
+  drawerSidebarContentInnerElement(2).should('have.text', 'link b')
+    .and('be.visible');
+  drawerSidebarContentInnerElement(3).should('have.text', 'link c')
+    .and('be.visible');
+});
+
+Then('sidebar text is not visible', () => {
+  drawerSidebarContentInnerElement(1).should('have.text', 'link a')
+    .and('not.be.visible');
+  drawerSidebarContentInnerElement(2).should('have.text', 'link b')
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/drawer-steps.js:29]:
```suggestion
  drawerSidebarContentInnerElement(positionOfElement('third')).should('have.text', 'link b')
```

File: cypress/support/step-definitions/drawer-steps.js [hunk 45]
```diff
@@ -0,0 +1,49 @@
...
+  drawerSidebarContentInnerElement(1).should('have.text', 'link a')
+    .and('be.visible');
+  drawerSidebarContentInnerElement(2).should('have.text', 'link b')
+    .and('be.visible');
+  drawerSidebarContentInnerElement(3).should('have.text', 'link c')
+    .and('be.visible');
+});
+
+Then('sidebar text is not visible', () => {
+  drawerSidebarContentInnerElement(1).should('have.text', 'link a')
```
Comment (KatarzynaQA) [inline cypress/support/step-definitions/drawer-steps.js:27]:
```suggestion
  drawerSidebarContentInnerElement(positionOfElement('second')).should('have.text', 'link a')
```

File: cypress/features/regression/test/drawer.feature (hunk 37 above)
Comment (damianOfKyan) [reply cypress/features/regression/test/drawer.feature]: c22cc5b

File: cypress/features/regression/test/drawer.feature (hunk 38 above)
Comment (damianOfKyan) [reply cypress/features/regression/test/drawer.feature]: c22cc5b

File: cypress/features/regression/test/drawer.feature (hunk 39 above)
Comment (damianOfKyan) [reply cypress/features/regression/test/drawer.feature]: c22cc5b

File: cypress/features/regression/test/drawer.feature (hunk 40 above)
Comment (damianOfKyan) [reply cypress/features/regression/test/drawer.feature]: c22cc5b

File: cypress/support/step-definitions/drawer-steps.js (hunk 41 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: c22cc5b

File: cypress/support/step-definitions/drawer-steps.js (hunk 42 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: c22cc5b

File: cypress/support/step-definitions/drawer-steps.js (hunk 43 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: c22cc5b

File: cypress/support/step-definitions/drawer-steps.js (hunk 44 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: c22cc5b

File: cypress/support/step-definitions/drawer-steps.js (hunk 45 above)
Comment (damianOfKyan) [reply cypress/support/step-definitions/drawer-steps.js]: c22cc5b

File: cypress/features/regression/test/drawer.feature [hunk 46]
```diff
@@ -0,0 +1,43 @@
...
+
+  @positive
+  Scenario Outline: Set animationDuration to <animationDuration>
+    Given I set animationDuration to "<animationDuration>"
+    When I click on Drawer arrow 1 time
+    Then animationDuration is set to "<cssAnimationDuration>"
+    Examples:
+      | animationDuration | cssAnimationDuration |
+      | 500ms          | 0.5s              |
+      | 100ms          | 0.1s              |
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/drawer.feature:29]: Please, format document to remove misaligned

File: cypress/features/regression/test/drawer.feature [hunk 47]
```diff
@@ -0,0 +1,43 @@
...
+      | 0.1s           | 0.1s              |
+
+  @positive
+  Scenario Outline: Set expandedWidth to <expandedWidth>
+    Given I set expandedWidth to "<expandedWidth>"
+    When I click on Drawer arrow 1 time
+    Then expandedWidth is set to "<cssWidth>"
+    Examples:
+      | expandedWidth                   | cssWidth      |
+      | 10%                             | 114.59375px   |
```
Comment (KatarzynaQA) [inline cypress/features/regression/test/drawer.feature:40]: Please, format document to remove empty space

File: cypress/features/regression/test/drawer.feature (hunk 46 above)
Comment (damianOfKyan) [reply cypress/features/regression/test/drawer.feature]: f13b199

File: cypress/features/regression/test/drawer.feature (hunk 47 above)
Comment (damianOfKyan) [reply cypress/features/regression/test/drawer.feature]: f13b199

File: src/components/drawer/drawer.stories.js [hunk 48]
```diff
@@ -0,0 +1,72 @@
...
+  const onChangeHandler = useCallback(() => {
+    setIsExpanded(!isExpanded);
+    action('expansionToggled');
+  }, [isExpanded]);
+
+  return (
+    <div>
+      <p>
+        Note: if you experience glitchy animation on `Drawer` component,
+        pelase open canvas in new window (2nd icon in top right corner)
```
Comment (ianforster-sage) [inline src/components/drawer/drawer.stories.js:51]:
```suggestion
        please open canvas in new window (2nd icon in top right corner)
```

**Pull request merged**